app/models/dialog_field_text_box.rb

```diff
@@ -1,9 +1,10 @@
 class DialogFieldTextBox < DialogField
-  AUTOMATE_VALUE_FIELDS = %w(protected required validator_rule validator_type read_only)
+  AUTOMATE_VALUE_FIELDS = %w(data_type protected required validator_rule validator_type read_only).freeze
 
   def value
     @value = values_from_automate if dynamic && @value.blank?
-    @value
+    return @value if @value.nil?
+    data_type == "integer" ? @value.to_i : @value
```

Comment:
Yeah, that does work, but I feel like it's kinda nice to have the explicitness of the nil being returned. Otherwise, it just falls through and relies on nothing happening in the method, which is hard to tell at a glance what is going on. Maybe changing it to explicitly returning `nil` instead of `@value` would be more clear, even though as you pointed out it's technically not even needed.

@gmcculloug Thoughts?